### PR TITLE
[jobs] Add /api/version endpoint

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

- Allows us to throw a useful error message when a user is trying to run job submission against an old version of Ray.
- Allows us to detect version incompatibilities in the future by checking the Ray version / commit SHA + a version number we can increment.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/job/job_head.py [hunk 1]
```diff
@@ -69,6 +71,18 @@ def job_exists(self, job_id: str) -> bool:
         status = self._job_manager.get_job_status(job_id)
         return status is not None
 
+    @routes.get("/api/version")
+    async def get_version(self, req: Request) -> Response:
+        resp = VersionResponse(
+            version=CURRENT_VERSION,
```
Comment (jiaodong) [inline dashboard/modules/job/job_head.py:77]: can we add a comment here suggesting the value here is temporary and hardcoded as 0 for now, but subject to change later on ?

File: dashboard/modules/job/job_head.py [hunk 2]
```diff
@@ -69,6 +71,18 @@ def job_exists(self, job_id: str) -> bool:
...
+            ray_version=ray.__version__,
```
Comment (jiaodong) [inline dashboard/modules/job/job_head.py:78]: just for curiosity .. is it possible or supported that user submits a job with a different ray version / commit, such that laptop sdk has different version compare to the remote ray cluster ? It looks like a common case at least at commit level later on.

**Review by jiaodong — APPROVED** [review]

File: dashboard/modules/job/job_head.py (hunk 2 above)
Comment (edoakes) [reply dashboard/modules/job/job_head.py:78]: Yes definitely. We should try to maintain compatibility as much as possible. It's a big pain point right now for Ray client that you need to match basically the exact version.

**Comment by edoakes** [comment]
Only flaky test failures

**Pull request merged**